### DISCUSSION TITLE
Preallocated BodyID

### DISCUSSION THREAD

**Discussion by AndreaCatania** [jrouwe/JoltPhysics#697] [Q&A] [answered]
Hello, The scene I'm working on has a mix of networked and non-networked bodies.
In order to properly network the physics, it's necessary that the peers share the same body IDs; I would like to reserve the first 1 thousand BodyIDs for the non networked bodies, so that each client can spawn the non networked bodies without worrying too much about breaking the sync as the first thousand BodyIDs will never be sync.

How can I reserve some `BodyID`s?

Also, is it possible to get a body by fetching it using "just" the `index` (ignoring the `sequence`)? I need this to verify if an index is already allocated.

**Comment by jrouwe** [comment, marked as answer]
> How can I reserve some BodyIDs?

There is currently no way of reserving body IDs, basically if you want to hand out your own body IDs you should hand out ALL body IDs yourself or you risk that the system allocates an ID that you want to hand out. I would suggest keeping track of a simple bit array to know which bodies you handed out.

> Also, is it possible to get a body by fetching it using "just" the index (ignoring the sequence)? I need this to verify if an index is already allocated.

If you call ```BodyInterface::AssignBodyID``` with a body ID that is already in use (possibly using a different sequence number) it will return false.

**Reply by AndreaCatania** [reply]
Ok, it sounds good, thank you :)
